File: Backend/StockDaddy.API/Application/DTOs/SaleDto.cs
using StockDaddy.Domain.Enums;

namespace StockDaddy.Application.DTOs;

public class SaleDto
{
    public int Id { get; set; }
    public int TenantId { get; set; }
    public int? StoreId { get; set; }
    public int? CustomerId { get; set; }
    public int? CompanyId { get; set; }
    public int SoldBy { get; set; }
    public string? SoldByName { get; set; }

    public decimal TotalAmount { get; set; }
    public decimal SubtotalAmount { get; set; }
    public decimal TaxAmount { get; set; }
    public decimal DiscountAmount { get; set; }
    public PaymentMethod PaymentMethod { get; set; }

    public string? CustomerName { get; set; }
    public string? CompanyName { get; set; }
    public string? Notes { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: Frontend/src/dtos/sale.dto.ts
export type PaymentMethod = 'Cash' | 'Card' | 'UPI' | 'BankTransfer' | 'Credit';

export interface SaleItemDto {
  id: number;
  saleId: number;
  productVariantId: number;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
}

export interface SaleDto {
  id: number;
  tenantId: number;
  storeId?: number;
  customerId?: number;
  customerName?: string;
  companyId?: number;
  companyName?: string;
  soldBy: number;
  soldByName?: string;
  subtotalAmount?: number;
  taxAmount?: number;
  discountAmount?: number;
  totalAmount: number;
  paymentMethod: PaymentMethod;
  notes?: string;
  createdAt: string;
  updatedAt: string;
  saleItems?: SaleItemDto[];
}

export interface CreateSaleItemRequest {
  saleId: number;
  productVariantId: number;
  quantity: number;
  unitPrice: number;
}

export interface CreateSaleRequest {
  tenantId: number;
  storeId?: number;
  customerId?: number;
  soldBy: number;
  totalAmount: number;
  paymentMethod: PaymentMethod;
  notes?: string;
}

export interface UpdateSaleRequest {
  totalAmount: number;
  paymentMethod: PaymentMethod;
  notes?: string;
}

export interface InvoiceDto {
  id: number;
  saleId: number;
  invoiceNumber: string;
  invoiceDate: string;
  dueDate: string;
  status: string;
  fileUrl?: string;
}
